spyder/widgets/findinfiles.py

```diff
@@ -487,32 +574,30 @@ def get_options(self, all=False):
         else:
             return (path, file_search, exclude, texts, text_re, case_sensitive)
 
-    @Slot()
-    def select_directory(self):
-        """Select directory"""
-        self.parent().redirect_stdio.emit(False)
-        directory = getexistingdirectory(self, _("Select directory"),
-                                         self.path)
-        if directory:
-            directory = to_text_string(osp.abspath(to_text_string(directory)))
-        self.parent().redirect_stdio.emit(True)
-        return directory
+    @property
+    def path(self):
+        return self.path_selection_combo.path
 
     def set_directory(self, directory):
-        self.path = to_text_string(osp.abspath(to_text_string(directory)))
+        self.path_selection_combo.path = to_text_string(osp.abspath(
+                to_text_string(directory)))

```

Comment:
Should `to_text_string(osp.abspath(to_text_string(directory)))` be changed to `osp.abspath(directory)` or `to_unicode_from_fs(osp.abspath(path))`?

Reply:
I see this is only used in one place:

https://github.com/jnsebgosselin/spyder/blob/8b9483d72967c4580143fde6adaeec38704f0dbd/spyder/plugins/findinfiles.py#L78

So this should be `osp.abspath(directory)` because `getcwd_or_home` already returns unicode.

Reply:
Ok thank you. I'll remove the `to_text_string` here.